### ISSUE TITLE
Feature request: Return Transaction Hash instead of waiting for Tx Inclusion

### ISSUE THREAD

**Issue by insipx** [xmtp/xps-gateway#52]
### Is your feature request related to a problem?

for context: https://github.com/xmtp/xps-gateway/pull/50#discussion_r1470035231_

We currently wait on `PendingTransaction` until it is included, instead we can return a transaction hash and let the client wait

**Comment by 37ng** [comment]
I think the gateway shouldn't return tx hash, it should limit the blockchain logic within it. However it's always possible for a tx to fail and it's the gateway who should take care of monitor / resend logic instead of its client.

So I'm proposing a "two-checkmarks" mechanism to solve this problem:
1st ☑️ happens when the gateway received the request and sent a tx to blockchain, without waiting, it returns the result with a `pending` status.

2nd ☑️ happens when the gateway confirmed that the tx is included. Then the gateway update the status to `success` or `failed` depending on the blockchain response. If the tx is lost due to network issue, the gateway should resend it.

**Comment by insipx** [comment]
Yeah, that makes sense. We would basically be emulating the ethers api -- we could even just send a struct wrapping PendingTransaction, which is a future that just polls the PendingTransaction future

this would then resolve to success or failure

**Comment by 37ng** [comment]
> we could even just send a struct wrapping PendingTransaction

Do you mean gateway sends a `PendingTransaction` back to its client? Would that mean the client would need to deal with blockchain specifics?

I think it's important for gateway to abstract away all the blockchain details. It only tells the client simple responses like `success`, `pending` or `failure`.

**Comment by insipx** [comment]
Not necessarily, we would still have control over the blockchain specifics via visibility. I was thinking of something like this:

```rust
pub struct Pending {
    ethers_pending: PendingTransaction
}

impl Future for Pending {
    fn poll(self: &mut Pin<Self>) -> Result<PollResult> {
        self.ethers_pending.poll()
    }
}
```

Client-side
```rust

// make a request and get back our pending struct
let pending = client.grant_installation(/* .. */).await?;
let result = pending.await;
assert_eq!(pending.status, Status::Success);
```

**Comment by 37ng** [comment]
I agree, this is more clear. And appreciate the sample code, that's very helpful.
